### PR TITLE
Issue/166 async correlation

### PR DESCRIPTION
**Pull request by szpak**
- Tests for correlationId propagation in asynchronous calls.
 - More configurable tests hierarchy

Btw, do you have a good idea how to make base Spec class with CorrelationIdFilter in version with and without Wiremock without duplicating a class hierarchy?

### CODE REVIEW

**Comment by szpak** [comment]
Hopefully this PR is ready for review.

File: micro-deps/micro-deps-spring-test-config/src/main/groovy/com/ofg/infrastructure/base/MvcWiremockIntegrationSpec.groovy [hunk 1]
```diff
@@ -24,12 +24,12 @@ import static com.ofg.config.BasicProfiles.TEST
 @CompileStatic
 @ContextConfiguration(classes = [MockServerConfiguration])
 @ActiveProfiles(TEST)
-class MvcWiremockIntegrationSpec extends MvcIntegrationSpec {
+abstract class MvcWiremockIntegrationSpec extends MvcIntegrationSpec {
+
+    @Autowired protected HttpMockServer httpMockServer
+    @Value('${wiremock.url:localhost}') protected String wiremockUrl
```
Comment (nurkiewicz) [inline micro-deps/micro-deps-spring-test-config/src/main/groovy/com/ofg/infrastructure/base/MvcWiremockIntegrationSpec.groovy]: This is no longer valid after https://github.com/4finance/micro-infra-spring/pull/196, just hard code `localhost`

Comment (szpak) [inline micro-deps/micro-deps-spring-test-config/src/main/groovy/com/ofg/infrastructure/base/MvcWiremockIntegrationSpec.groovy]: Ok, I will rebase this branch against master and change it.

File: micro-infra-spring-base/src/main/groovy/com/ofg/infrastructure/correlationid/CorrelationIdUpdater.groovy [hunk 2]
```diff
@@ -45,5 +48,78 @@ class CorrelationIdUpdater {
...
+     *         extractedLocations << locationExtractor.fromTweet(tweet)
+     *     }
+     * }
+     * </code></pre>
+     *
+     * <b>Note</b>: Passing only one input parameter currently is supported.
+     *
+     * @param block code block to execute in a thread with a correlation ID taken from the original thread
+     * @return wrapping block as Closure
+     * @since 0.8.4
```
Comment (nurkiewicz) [inline micro-infra-spring-base/src/main/groovy/com/ofg/infrastructure/correlationid/CorrelationIdUpdater.groovy]: +1 for documentation

File: micro-infra-spring-base/src/main/groovy/com/ofg/infrastructure/correlationid/CorrelationIdUpdater.groovy [hunk 3]
```diff
@@ -45,5 +48,78 @@ class CorrelationIdUpdater {
...
+     */
+    static <T> Closure<T> closureWithId(Closure<T> block) {
```
Comment (nurkiewicz) [inline micro-infra-spring-base/src/main/groovy/com/ofg/infrastructure/correlationid/CorrelationIdUpdater.groovy]: Suggested names: 
* `wrapWithCorrId()`
* `wrapWithCorrelationId()`
* `wrappedWithPresentCorrelationId()`

Comment (szpak) [inline micro-infra-spring-base/src/main/groovy/com/ofg/infrastructure/correlationid/CorrelationIdUpdater.groovy]: We also need to distinguish a version with `Closure` and with `Callable`.

File: micro-infra-spring-base/src/test/groovy/com/ofg/infrastructure/correlationid/CorrelationIdUpdaterTest.groovy [hunk 4]
```diff
@@ -25,12 +30,112 @@ class CorrelationIdUpdaterTest extends Specification {
...
+    }
+
+    def "should propagate correlation ID into nested Callable"() {
+        given:
+            ExecutorService threadPool = Executors.newFixedThreadPool(1)
+            CorrelationIdUpdater.updateCorrelationId('A')
+            Callable<String> callable = new CorrelationIdTestCallable()
+        when:
+            Callable<String> wrappedCallable = CorrelationIdUpdater.callableWithId(callable)
+            String nestedCorrelationId = threadPool.submit(wrappedCallable).get(1, TimeUnit.SECONDS)
```
Comment (nurkiewicz) [inline micro-infra-spring-base/src/test/groovy/com/ofg/infrastructure/correlationid/CorrelationIdUpdaterTest.groovy]: Static import of `TimeUnit.SECONDS` would improve readability.

**Comment by nurkiewicz** [comment]
[LGTM](http://acronyms.thefreedictionary.com/LGTM).

**Pull request merged**